lock.rst

```diff
@@ -0,0 +1,291 @@
+        }
+    }
+
+Named Lock
+----------
+
+If the application needs different kind of Stores alongside each other, Symfony
+provides :doc:`named lock <reference-lock-resources-name>`::
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/lock.yaml
+        framework:
+            lock:
+                invoice: ['semaphore', 'redis://r2.docker']
+                report: 'semaphore'
+
+    .. code-block:: xml
+
+        <!-- config/packages/lock.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:lock>
+                    <framework:resource name="invoice">semaphore</framework:resource>
+                    <framework:resource name="invoice">redis://r2.docker</framework:resource>
+                    <framework:resource name="report">semaphore</framework:resource>
+                </framework:lock>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/lock.php
+        $container->loadFromExtension('framework', [
+            'lock' => [
+                'invoice' => ['semaphore', 'redis://r2.docker'],
+                'report' => 'semaphore',
+            ],
+        ]);
+
+Each name becomes a service where the service id suffixed by the name of the
+lock (e.g. ``lock.invoice``). An autowiring alias is also created for each lock
+using the camel case version of its name suffixed by ``Lock`` - e.g. ``invoice``
+ can be injected automatically by naming the argument ``$invoiceLock`` and
+ type-hinting it with :class:`Symfony\\Component\\Lock\\LockInterface`.
+
+Symfony also provide a corresponding factory and store following the same rules
+(e.g. ``invoice`` generates a ``lock.invoice.factory`` and
+``lock.invoice.store``, both can be injected automatically by naming
+respectively ``$invoiceLockFactory`` and ``invoiceLockStore`` and type-hinted
+with :class:`Symfony\\Component\\Lock\\LockFactory` and
+:class:`Symfony\\Component\\Lock\\PersistingStoreInterface`)
+
+Blocking Store
+--------------
+
+If you want to use the ``RetryTillSaveStore`` for :ref:`non-blocking locks <lock-blocking-locks>`,
+you can do it by :doc:`decorating the store </service_container/service_decoration>` service:
+
+.. code-block:: yaml
+
+    lock.default.retry_till_save.store:
+        class: Symfony\Component\Lock\Store\RetryTillSaveStore
+        decorates: lock.default.store
+        arguments: ['@lock.default.retry.till.save.store.inner', 100, 50]
```

Comment:
I think there is a typo in the inner service name

```suggestion
        arguments: ['@lock.default.retry_till_save.store.inner', 100, 50]
```

Reply:
indeed. good catch